### ISSUE TITLE
ILSM: Analyze Interconnection Structure of Multilayer Interaction Networks

### ISSUE THREAD

**Issue by WeichengSun** [ropensci/software-review#654]
Submitting Author Name:  SunWeicheng
Submitting Author Github Handle: @WeichengSun
Repository:  https://github.com/WeichengSun/ILSM
Version submitted: 1.0.3.2
Submission type: Stats
Editor:  TBD 
Reviewers:  TBD 

Archive: TBD
Version accepted: TBD
Language: en

---

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: ILSM
Title: Analyze Interconnection Structure of Multilayer Interaction Networks
Version: 1.0.3.2
Authors@R: c(
    person("WeiCheng", "Sun", , "sunwch2023@lzu.edu.cn", role = c("aut", "cre")),
    person("Chuan", "Yan", , "yanchuan@lzu.edu.cn", role = c("aut")),
    person("Yangyang", "Zhao", , "120220907711@lzu.edu.cn", role = c("aut")))
Description: 
    In view of the analysis of the structural characteristics of the multilayer network has been complete, however, there is still a lack of a unified operation that can quickly obtain the corresponding characteristics of the multilayer network. 
    To solve this insufficiency, 'ILSM' was designed for supporting calculating such metrics of multilayer networks by functions of this R package.
License: MIT + file LICENSE
URL: https://github.com/WeichengSun/ILSM
BugReports: https://github.com/WeichengSun/ILSM/issues
Encoding: UTF-8
Roxygen: list(markdown = TRUE)
RoxygenNote: 7.3.1
Depends: R (>= 3.5.0)
Imports: 
    stats,
    Matrix,
    igraph
Suggests: 
    knitr,
    rmarkdown,
    testthat (>= 3.0.0)
VignetteBuilder: knitr
Config/testthat/edition: 3

```

## Scope

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) this package falls under: (Please check an appropriate box below. If you are unsure, we suggest you make a pre-submission inquiry.):

- Data Packages
	- [ ] data retrieval
	- [ ] data extraction
	- [ ] data munging
	- [ ] data deposition
    - [ ] data validation and testing
	- [ ] workflow automation
	- [ ] version control
	- [ ] citation management and bibliometrics
	- [ ] scientific software wrappers
	- [ ] field and lab reproducibility tools
	- [ ] database software bindings
	- [ ] geospatial data
	- [ ] text analysis
- Statistical Packages
     - [ ] bayesian and monte carlo routines
     - [ ] dimensionality reduction, clustering, and unsupervised learning
     - [ ] machine learning
     - [x] ecological network analysis
     - [ ] regression and supervised learning
     - [x] exploratory data analysis (EDA) and summary statistics
     - [ ] spatial analysis
     - [ ] time series analysis

- Explain how and why the package falls under these categories (briefly, 1-2 sentences):

This package was designed for supporting calculating such metrics of multilayer networks.

-   Who is the target audience and what are scientific applications of this package?
   
Reseachers who work for ecological interaction effects.

-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?

-   (If applicable) Does your package comply with our [guidance around _Ethics, Data Privacy and Human Subjects Research_](https://devguide.ropensci.org/policies.html#ethics-data-privacy-and-human-subjects-research)?

Yes!

-   If you made a pre-submission inquiry, please paste the link to the corresponding issue, forum post, or other discussion, or @tag the editor you contacted.

-   Explain reasons for any [`pkgcheck` items](https://docs.ropensci.org/pkgcheck/) which your package is unable to pass.

## Technical checks

Confirm each of the following by checking the box.

- [x] I have read the [rOpenSci packaging guide](https://devguide.ropensci.org/building.html).
- [x] I have read the [author guide](https://devdevguide.netlify.app/authors-guide.html) and I expect to maintain this package for at least 2 years or to find a replacement.

This package:

- [x] does not violate the Terms of Service of any service it interacts with.
- [x] has a CRAN and OSI accepted license.
- [x] contains a [README with instructions for installing the development version](https://github.com/WeichengSun/ILSM/blob/main/README.Rmd).
- [x] includes [documentation with examples for all functions, created with roxygen2](https://ropensci.github.io/dev_guide/building.html#documentation).
- [x] contains a vignette with examples of its essential functions and uses.
- [x] has a [test suite](https://github.com/WeichengSun/ILSM/tree/main/tests).
- [x] has [continuous integration](https://ropensci.github.io/dev_guide/ci.html), including reporting of test coverage.

## Publication options

- [x] Do you intend for this package to go on CRAN?
- [ ] Do you intend for this package to go on Bioconductor?

- [x] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:

<details>
<summary>MEE Options</summary>

- [x] The package is novel and will be of interest to the broad readership of the journal.
- [x] The manuscript describing the package is no longer than 3000 words.
- [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
- (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no guarantee that your manuscript will be within MEE scope.*)
- (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
- (*Please do not submit your package separately to Methods in Ecology and Evolution*)

</details>

## Code of conduct

- [x] I agree to abide by [rOpenSci's Code of Conduct](https://ropensci.org/code-of-conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by WeichengSun** [comment]
@ropensci-review-bot check it. Thanks!

**Comment by WeichengSun** [comment]
@ropensci-review-bot check package

**Comment by WeichengSun** [comment]
@ropensci-review-bot check package

**Comment by WeichengSun** [comment]
@ropensci-review-bot help

**Comment by WeichengSun** [comment]
@ropensci-review-bot check package

**Comment by WeichengSun** [comment]
@ropensci-review-bot check package

**Comment by adamhsparks** [comment]
Hi @WeichengSun, it seems that the bot is having some issues, I'll see what we can do about that. Do you still wish to proceed with the review?

If so, you'll need to set up [CI](https://ropensci.github.io/dev_guide/ci.html) with test coverage reporting and ensure all of your exported functions have examples in your [ROxygen code](https://ropensci.github.io/dev_guide/building.html#documentation) as you've not indicated that these criteria are met.

**Comment by WeichengSun** [comment]
Hey, @adamhsparks, thanks for your suggestion, I have revised part of it. Since this is the first submission, there may be many questions. Please give us more suggestions on the future progress.

**Comment by adamhsparks** [comment]
@ropensci-review-bot check package

**Comment by adamhsparks** [comment]
@WeichengSun, the package checks indicate that the test coverage needs to be improved before we can start the review but everything else checks out.

Would you like to proceed? If so, just ping me in this Issue when you've got the package's test coverage up to 75% and I'll take it from there. Also, feel free to ask me if you have any other questions.

**Comment by WeichengSun** [comment]
Hi @adamhsparks,  I've added some tests to address the test coverage issue, but I wonder why the report shows package name is not available. If so, you can continue. Thank you.

**Comment by adamhsparks** [comment]
You have it on CRAN, already, so it shows unavailable.

**Comment by adamhsparks** [comment]
@ropensci-review-bot check package

**Comment by adamhsparks** [comment]
@WeichengSun, could you please set up or fix the test coverage for GitHub Actions and add a badge for it? This indicates that it's not passing, https://github.com/WeichengSun/ILSM/actions/runs/10752142570

**Comment by WeichengSun** [comment]
Hi @adamhsparks, it's passing now.

**Comment by WeichengSun** [comment]
Hi @adamhsparks, I want to submit an Applications Article about my package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/), but I don't know how my package can pass the review of ropensci. Please give me some suggestions. Thank you.

**Comment by adamhsparks** [comment]
Hi @WeichengSun, I'm currently seeking a handling editor for your package but haven't found anyone yet. I'll get someone assigned as soon as I can.

**Comment by adamhsparks** [comment]
Hi @WeichengSun, after checking with the editorial board, I was alerted that you seemingly have mashed together our pre-submission and regular stats templates with your submission.

Could I ask you to please close this and open a regular stats submission so that the bot picks up this as a stats submission?

You will need to add our stats standard annotations to the package as well. The bot will pick that up and the process won't proceed until you've done that, so I suggest doing that before opening a new issue and using the proper stats template.

**Comment by WeichengSun** [comment]
Hi @adamhsparks, what can I refer to for 'stats standard annotations' you mentioned? I am very confused about this.

**Comment by adamhsparks** [comment]
Hi @WeichengSun, I'd start with our [rOpenSci Statistical Software Peer Review](https://stats-devguide.ropensci.org/) guide, specifically [Chapter3](https://stats-devguide.ropensci.org/pkgdev.html#general-standards) for these details.
